h/services/user_delete.py

```diff
@@ -46,6 +46,20 @@ def delete_user(self, user: User):
             else:
                 self._db.delete(group)
 
+        self._db.add(
+            UserDeletion(
+                userid=user.userid,
+                requested_by=requested_by.userid,
+                tag=tag,
+                created_at=user.registered_date,
+                num_annotations=self._db.scalar(
+                    sa.select(
+                        sa.func.count(Annotation.id)  # pylint:disable=not-callable
+                    ).where(Annotation.userid == user.userid)
+                ),
```

Comment:
Querying the annotation table here. Any chance we might want to move this to `annotation_slim`?

Reply:
Not yet (if you want to get the exact number), just created:

https://github.com/hypothesis/h/issues/8727 to finish that.